src/cpdetect/compiler_detector/heuristics/pe_heuristics.cpp

```diff
@@ -1142,6 +1142,8 @@ void PeHeuristics::getExcelsiorHeuristics()
  */
```

Comment:
_(A comment not related to the attached line.)_

The following regression tests have failed [in this TC build](https://retdec-tc.avast.com/viewLog.html?buildId=3716&buildTypeId=Retdec_LinuxBuild&tab=buildLog):
```
tools.fileinfo.features.section-name-packer-compiler-detection.TestRamnit (b9bf69972077eee9c6[..])  [ FAIL ]  (0.14s)
tools.fileinfo.features.vmprotect-heuristic.Test (TidyESP.ex)                                       [ FAIL ]  (0.22s)
```
Could you please take a look at them? The failures seem to be caused by the changes in the present PR.

Reply:
You can run them in the following way:
```
$ ./runner.py tools.fileinfo.features.section-name-packer-compiler-detection -r TestRamnit
$ ./runner.py tools.fileinfo.features.vmprotect-heuristic -r TidyESP
```

Reply:
The second test has been fixed. As for the first test, it fails only in TeamCity and only in a debug Linux build. For some reason, `fileinfo.return_code` is 245. I have no idea why. I will have to dig deeper.

Reply:
It was probably caused by the following code:

```
auto data = section->getBytes().data();
```
The `.data()` returns a reference to already existing data, but that data gets destroyed after that code finished executing and the reference is invalidated.

To fix this I have moved the result of the `section->getBytes()` to a separate variable, which will only be destroyed when its scope ends.

Reply:
This did not fix the crashing. For more details, see [this comment](https://github.com/avast/retdec/pull/734#issuecomment-611977060). All the tests pass now, so I am marking this as resolved.